code/src/lib.rs

```diff
@@ -0,0 +1,209 @@
+use proc_macro::TokenStream;
+use unscanny::Scanner;
+
+/// Interpolation kind
+enum Pattern<'a> {
+    Display(&'a str),
+    Iterator(&'a str),
+    Call(&'a str),
+}
+
+/// Match allowed character at the beginning of a Rust identifier
+fn ident_start(c: char) -> bool {
+    c.is_alphabetic() || c == '_'
+}
+
+/// Match allowed character after the beginning of a Rust identifier
+fn ident_body(c: char) -> bool {
+    c.is_alphanumeric() || c == '_'
+}
```

Comment:
For full compliance, this should probably use `is_xid_start` and `is_xid_continue` from https://github.com/unicode-rs/unicode-xid. AFAIK, this is how rust identifiers are defined

```rust
fn is_ident_start(c: char) -> bool {
    unicode_xid::UnicodeXID::is_xid_start(c) || c == '_'
}

fn is_ident_continue(c: char) -> bool {
    unicode_xid::UnicodeXID::is_xid_continue(c)
}
```

Note that `is_xid_start` does not include underscores (but they are valid rust identifier start), while `is_xid_continue` does include them.